src/Agda2Hs/Compile/Utils.hs

```diff
@@ -239,3 +239,12 @@ addTyBang :: Strictness -> Hs.Type () -> C (Hs.Type ())
 addTyBang Strict ty = tellExtension Hs.BangPatterns >>
   return (Hs.TyBang () (Hs.BangedTy ()) (Hs.NoUnpackPragma ()) ty)
 addTyBang Lazy   ty = return ty
+
+checkSingleField :: Hs.Name () -> [b] -> C ()
+checkSingleField name fs = unless (length fs == 1) $ genericDocError =<< do
+  text "Newtype must have exactly one field in definition: " <+> text (Hs.prettyPrint name)
+
+checkValidNewtype :: Hs.Name () -> Hs.Name () -> [b] -> C ()
+checkValidNewtype cName rName fs = do
+  checkValidConName cName
+  checkSingleField rName fs
```

Comment:
If this not used anywhere, please delete. If you decide to keep, use it in the `ToRecordNewType` target above, but change the name to specify that it only refers to record newtypes.

Reply:
Removed it!